java/org/apache/catalina/servlets/DefaultServlet.java

```diff
@@ -1650,7 +1654,7 @@ protected InputStream renderXml(HttpServletRequest request, String contextPath,
               .append('\'');
             sb.append(" urlPath='")
               .append(rewrittenContextPath)
-              .append(rewriteUrl(directoryWebappPath + entry))
+              .append(rewriteUrl(directoryWebappPath + entry, URLEncoder.XML))
```

Comment:
using XML encoder to encode path attribute which may contain "'" and "&" symbols